src/Consolonia.Core/Text/Fonts/AsciiArtTypeface.cs

```diff
@@ -0,0 +1,405 @@
+
+        /// <summary>
+        ///     Check if two characters can smush together, returning the resulting character or '\0' if they can't
+        /// </summary>
+        private char TrySmush(char leftChar, char rightChar)
+        {
+            // If either is space, they can't smush (but can kern)
+            if (leftChar == ' ' || rightChar == ' ')
+                return '\0';
+
+            // Hardblank is treated specially
+            if (leftChar == Hardblank || rightChar == Hardblank)
+            {
+                if (LayoutMode.HasFlag(LayoutMode.Hardblank))
+                    return leftChar == Hardblank ? rightChar : leftChar;
+                return '\0';
+            }
+
+            // Check smush rules
+            if (LayoutMode.HasFlag(LayoutMode.Equal))
+                if (leftChar == rightChar)
+                    return leftChar;
+
+            if (LayoutMode.HasFlag(LayoutMode.Lowline))
+            {
+                if (leftChar == '_' && "|/\\[]{}()<>".Contains(rightChar))
+                    return rightChar;
+                if (rightChar == '_' && "|/\\[]{}()<>".Contains(leftChar))
+                    return leftChar;
+            }
+
+            if (LayoutMode.HasFlag(LayoutMode.Hierarchy))
+            {
+                string hierarchy = "|/\\[]{}()<>";
+                int leftLevel = hierarchy.IndexOf(leftChar);
+                int rightLevel = hierarchy.IndexOf(rightChar);
+
+                if (leftLevel >= 0 && rightLevel >= 0)
+                    return leftLevel > rightLevel ? leftChar : rightChar;
+            }
+
+            if (LayoutMode.HasFlag(LayoutMode.Pair))
+            {
+                if (leftChar == '[' && rightChar == ']' || leftChar == ']' && rightChar == '[')
+                    return '|';
+                if (leftChar == '{' && rightChar == '}' || leftChar == '}' && rightChar == '{')
+                    return '|';
+                if (leftChar == '(' && rightChar == ')' || leftChar == ')' && rightChar == '(')
+                    return '|';
+            }
+
+            if (LayoutMode.HasFlag(LayoutMode.BigX))
+            {
+                if (leftChar == '/' && rightChar == '\\')
+                    return '|';
+                if (leftChar == '\\' && rightChar == '/')
+                    return 'Y';
+                if (leftChar == '>' && rightChar == '<')
+                    return 'X';
+            }
+
+            return '\0';
+        }
+
+        // Resharper disable AssignNullToNotNullAttribute
+        public bool TryGetGlyphMetrics(ushort glyph, out GlyphMetrics metrics)
+        {
+            metrics = new GlyphMetrics
+            {
+                XBearing = 0,
+                YBearing = 0,
+                Height = Metrics.DesignEmHeight,
+                Width = GetGlyphAdvance(glyph)
+            };
+            return true;
+        }
+
+        public bool TryGetTable(uint tag, out byte[] table)
+        {
+            table = null;
```

Comment:
Do you know why resharper complains here since we don't have nullable context enabled here?

Reply:
Oh, it seems its enabled in Avalonia? If complain is only about this line just suppress it with `null!`

Reply:
yeah, assigning null to table, I changed to Array.Empty<byte>()